app/controllers/place_controller.rb

```diff
@@ -0,0 +1,93 @@
+require "slimmer/headers"
+
+class PlaceController < ApplicationController
+  before_filter :redirect_if_api_request
+  before_filter -> { set_expiry unless viewing_draft_content? }
+
+  INVALID_POSTCODE_ERROR = "invalidPostcodeError".freeze
+  NO_LOCATION_ERROR = "validPostcodeNoLocation".freeze
+
+  CUSTOM_SLUGS = {
```

Comment:
Might this be a lot simpler if it was handled in `routes.rb` and given it's own controller method?